waspc/src/Analyzer/TypeDefinitions.hs

```diff
@@ -0,0 +1,137 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Analyzer.TypeDefinitions
+  ( TypeDefinitions,
+    empty,
+    getDeclType,
+    getEnumType,
+    addDeclType,
+    addEnumType,
+    DeclType (..),
+    EnumType (..),
+    IsDeclType,
+    IsEnumType,
+  )
+where
+
+import Analyzer.Type (Type)
+import qualified Data.HashMap.Strict as M
+
+data EnumType = EnumType {etName :: String, etVariants :: [String]}
+
+data DeclType = DeclType {dtName :: String, dtBodyType :: Type}
+
+-- | Defines the declaration types and enum type that Analyzer is to take into
+--   consideration
+--
+--   The are used during type-checking and for constructing instances of
+--   Haskell types that correspond to these declaration and enums.
```

Comment:
```suggestion
-- | Parts of analyzer like parser (+ lexer), typechecker and evaluator are aware of declaration types, enum types
--   and quoter types as concepts, but they don't have the information on exact specific declaration, enum
--   and quoter types that are to be used during analysis.
--   This information is injected into them via 'TypeDefinitions', which defines in one place those specific
--   declaration/enum/quoter types.
--   This enables us to easily modify / add / remove specific types as Wasp evolves as a language without
--   having to touch the core functionality.

Defines the declaration types and enum type that Analyzer is to take into
--   consideration
--
--   The are used during type-checking and for constructing instances of
--   Haskell types that correspond to these declaration and enums.
```

Reply:
Thanks, that is better! I think we can remove last two lines "The type definitions are used ... ... Wasp file.", that was just a reminder of the old text which I forgot to remove, I don't think they add much.